### PR TITLE
Add trivial improvements to transaction pool

### PR DESCRIPTION
**Pull request by liuchengxu**
The changes in this PR are all trivial improvements to the client transaction pool:

- Use `Self`.
- Replace a `loop` with `while let`.
- Simplify the `Debug` impl for tags.
- Avoid `clone()` on `Copy` element.

### CODE REVIEW

**Review by NikVolf — APPROVED** [review]

File: client/transaction-pool/graph/src/future.rs [hunk 1]
```diff
@@ -47,24 +47,18 @@ impl<Hash: fmt::Debug, Ex: fmt::Debug> fmt::Debug for WaitingTransaction<Hash, E
...
-		let mut it = self.missing_tags.iter().map(|tag| HexDisplay::from(tag));
-		if let Some(tag) = it.next() {
-			write!(fmt, "{}", tag)?;
-		}
-		for tag in it {
-			write!(fmt, ", {}", tag)?;
-		}
-		write!(fmt, " }}}}")
+		write!(fmt, "missing_tags: {{{}}}", self.missing_tags.iter()
+			.map(|tag| HexDisplay::from(tag).to_string()).collect::<Vec<_>>().join(", "))?;
```
Comment (bkchr) [inline client/transaction-pool/graph/src/future.rs:51]:
```suggestion
		write!(
			fmt, 
			"missing_tags: {{{}}}", 
			self.missing_tags.iter()
				.map(|tag| HexDisplay::from(tag).to_string()).collect::<Vec<_>>().join(", "),
		)?;
```

File: client/transaction-pool/graph/src/base_pool.rs [hunk 2]
```diff
@@ -174,16 +174,9 @@ impl<Hash, Extrinsic> fmt::Debug for Transaction<Hash, Extrinsic> where
...
-			if let Some(t) = it.next() {
-				write!(fmt, "{}", HexDisplay::from(t))?;
-			}
-			for t in it {
-				write!(fmt, ",{}", HexDisplay::from(t))?;
-			}
-			Ok(())
-		}
+		let join_tags = |tags: &[Tag]| {
+			tags.iter().map(|tag| HexDisplay::from(tag).to_string()).collect::<Vec<_>>().join(",")
```
Comment (bkchr) [inline client/transaction-pool/graph/src/base_pool.rs:178]:
```suggestion
			tags.iter().map(|tag| HexDisplay::from(tag).to_string()).collect::<Vec<_>>().join(", ")
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
@liuchengxu could you please fix the failing tests.

**Comment by bkchr** [comment]
Ty

bot merge

**Comment by bkchr** [comment]
bot merge

**Pull request merged**